app/src/org/commcare/activities/CommCareHomeActivity.java

```diff
@@ -1098,6 +1098,10 @@ private boolean checkAndStartUnsentFormsTask(final boolean syncAfterwards,
     @Override
     protected void onResume() {
         super.onResume();
+        if (redirectedToLogin) {
+            finish();
+            return;
```

Comment:
We could also reuse this pattern for `onResume` as well: https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/activities/SessionAwareCommCareActivity.java#L18-L22